ClientGUI/ClientGUICreator.cs

```diff
@@ -20,5 +16,17 @@ public ClientGUICreator()
             AddControl(typeof(XNALinkButton));
             AddControl(typeof(XNAExtraPanel));
         }
+
+        private static ClientGUICreator _instance;
+        public static ClientGUICreator Instance
+        {
+            get
+            {
+                if (_instance == null)
+                    _instance = new ClientGUICreator();
+
+                return _instance;
+            }
+        }
```

Comment:
```suggestion
        private static ClientGUICreator _instance;
        public static ClientGUICreator Instance
        {
            get => _instance ??= new ClientGUICreator();
        }
```

Reply:
What are other's opinions? I think singleton is a pretty standard pattern so the one-line version is pretty intuitive even though it may be a bit complex logic-wise, and null-coalescing is a widely used thing anyways so it shouldn't bring readability issues.